packages/opentelemetry-context-async-hooks/src/AbstractAsyncHooksContextManager.ts

```diff
@@ -62,13 +67,13 @@ export abstract class AbstractAsyncHooksContextManager
       return this._bindEventEmitter(context, target);
     }
 
-    if (typeof target === 'function') {
+    if (isUnknownFunc(target)) {
```

Comment:
This fixes the lint error technically but doesn't actually fix the problem the lint is trying to point out. This will still return true for constructors so doesn't provide any additional safety. IMO it would be better to use a comment to disable the lint warning as a much more obvious solution.

Reply:
There are some workarounds to check if something is a constructor: https://stackoverflow.com/questions/40922531/how-to-check-if-a-javascript-function-is-a-constructor
Perhaps it would better also checking if the argument on `inUnknownFunc`/`isGenericFunc` is a constructor and, if yes, returns `false`?
I would like to avoid a comment ignoring the lint error.

Reply:
A comment ignoring the lint error clearly shows "i know this is dangerous and I accept it." A type which obfuscates the danger can come back to bite you later. If a constructor is passed to bind, there really isn't much we can or should do about it but let it fail.

Reply:
As far as I know NodeJs uses internally checks like `typeof(arg) === 'function'` and they don't care about constructors. Technically it's possible to create a function which can act as constructor _and_ as normal function.

Therefore I don't think we should add a lot complexity here trying to fix a problem which is on caller side.

Reply:
I agree with @Flarna here, i don't think we had the issue with any instrumentation so adding this much complexity for a problem that we don't have seems pointless right now. I would prefer to wait for an user having this issue to try to fix it